tidepool_data_science_models/models/icgm_sensor_generator_functions.py

```diff
@@ -189,10 +241,33 @@ def generate_icgm_sensors(
     #     iCGM = ((true_matrix + bias_matrix) * drift_multiplier) + noise
 
     # add delay or lag to the iCGM traces
-    delay_steps = np.int(np.round(delay / 5))
-    delayed_iCGM = np.insert(
-        values=iCGM[:, 0:1], obj=np.zeros(delay_steps, dtype=int), arr=iCGM[:, :-delay_steps], axis=1
-    )
+    if delay > 0:
+        delay_steps = np.int(np.round(delay / 5))
+        delayed_iCGM = np.insert(
+            values=iCGM[:, 0:1], obj=np.zeros(delay_steps, dtype=int), arr=iCGM[:, :-delay_steps], axis=1
+        )
+    else:
+        delayed_iCGM = iCGM
+
+
+    # add spurious events to the trace
+    # TODO: add in successive spurious events later, for now each spurious event is 5 minutes and can occur at any time
+    if number_of_spurious_events_per_10_days > 0:
+
+        true_matrix_shape = true_matrix.shape
+        n_icgm_data_points = true_matrix_shape[1]
+
+        state = np.zeros(shape=true_matrix_shape, dtype=np.int)
+        spurious_index = np.random.randint(
+            0, n_icgm_data_points, (n_sensors, int(number_of_spurious_events_per_10_days))
```

Comment:
I *think* I understand now how this is implemented. I also think we may need/want more flexibility now and likely would in the future. For example, as I understand it, a batch of 30 sensors can _only_ have [0, 30, 60, 90, ...,] spurious events in the collective traces. Or in an extreme example, a batch of 1000 sensors can _only_ have [0, 1000, 2000, ...] spurious events. We can't have say 5 events for 30 sensors or 500 events for 1000 sensors. Is that right? If so, is there a reason why it wasn't set it up to be parameterized to any number of spurious events for any batch size, which is pretty straightforward? The potential issue I see right now with this is it could affect the brute force fitting, and in the future we may want more control for other applications. @ed-nykaza